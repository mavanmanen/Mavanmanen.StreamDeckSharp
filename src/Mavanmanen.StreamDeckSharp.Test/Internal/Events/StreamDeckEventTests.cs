using System.IO;
using System.Reflection;
using System.Threading.Tasks;
using ApprovalTests.Reporters;
using Mavanmanen.StreamDeckSharp.Internal.Events;
using Mavanmanen.StreamDeckSharp.Test.TestHelpers;
using Xunit;
using Xunit.Abstractions;

namespace Mavanmanen.StreamDeckSharp.Test.Internal.Events
{
    [UseReporter(typeof(DiffReporter))]
    public class StreamDeckEventTests : XunitApprovalBase
    {
        public StreamDeckEventTests(ITestOutputHelper output) : base(output)
        {
        }

        private static async Task<string> GetEmbeddedJsonAsync(EventTypes eventType)
        {
            await using Stream stream = Assembly
                .GetExecutingAssembly()
                .GetManifestResourceStream($"Mavanmanen.StreamDeckSharp.Test.Internal.Events.Input.{eventType:G}.json");

            using var streamReader = new StreamReader(stream!);
            return await streamReader.ReadToEndAsync();
        }

        [Theory]
        [InlineData(EventTypes.KeyDown)]
        [InlineData(EventTypes.KeyUp)]
        [InlineData(EventTypes.WillAppear)]
        [InlineData(EventTypes.WillDisappear)]
        [InlineData(EventTypes.TitleParametersDidChange)]
        [InlineData(EventTypes.DeviceDidConnect)]
        [InlineData(EventTypes.DeviceDidDisconnect)]
        [InlineData(EventTypes.ApplicationDidLaunch)]
        [InlineData(EventTypes.ApplicationDidTerminate)]
        [InlineData(EventTypes.SendToPlugin)]
        [InlineData(EventTypes.DidReceiveSettings)]
        [InlineData(EventTypes.DidReceiveGlobalSettings)]
        [InlineData(EventTypes.PropertyInspectorDidAppear)]
        [InlineData(EventTypes.PropertyInspectorDidDisappear)]
        [InlineData(EventTypes.SystemDidWakeUp)]
        private async void FromJson_WithInput_ParsesCorrectly(EventTypes eventType)
        {
            // Arrange
            string json = await GetEmbeddedJsonAsync(eventType);

            // Act
            StreamDeckEvent streamDeckEvent = StreamDeckEvent.FromJson(json);

            // Assert
            Assert.NotNull(streamDeckEvent);
            Assert.Equal(eventType, streamDeckEvent.Event);
            StreamDeckApprovals.VerifyObject(streamDeckEvent);
        }
    }
}
