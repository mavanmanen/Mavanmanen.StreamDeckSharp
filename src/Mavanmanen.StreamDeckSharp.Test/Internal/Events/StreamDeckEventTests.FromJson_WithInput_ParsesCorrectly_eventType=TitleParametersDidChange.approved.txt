{"payload":{"settings":null,"coordinates":{"column":3,"row":1},"state":0,"title":"","titleParameters":{"fontFamily":"Arial","fontSize":12,"fontStyle":"Regular","fontUnderline":false,"showTitle":true,"titleAlignment":"Bottom","titleColor":"#ffffff"}},"action":"com.elgato.example.action1","context":"opaqueValue","device":"opaqueValue","event":4}